src/devno.rs

```diff
@@ -9,6 +9,30 @@ use crate::err::Result;
 /// Device number
 pub struct BlkidDevno(libc::dev_t);
 
+#[cfg(target_os = "linux")]
+pub mod ty {
+    /// Type alias for device major number
+    #[allow(non_camel_case_types)]
+    pub type maj_t = libc::c_uint;
+
+    /// Type alias for device minor number
+    #[allow(non_camel_case_types)]
+    pub type min_t = libc::c_uint;
+}
+
+#[cfg(all(unix, not(target_os = "linux"), not(target_os = "android")))]
+pub mod ty {
+    /// Type alias for device major number
+    #[allow(non_camel_case_types)]
+    pub type maj_t = i32;
+
+    /// Type alias for device minor number
+    #[allow(non_camel_case_types)]
+    pub type min_t = i32;
```

Comment:
I just wanted to double check the documentation for MacOS

https://developer.apple.com/documentation/kernel/1547212-makedev
https://developer.apple.com/documentation/kernel/1547203-major

I'm not certain, but it appears that the input should theoretically be unsigned while the output should be signed.

Reply:
Well, isn't that interesting... That means there's already a mismatch between the apple docs and the Rust implementation in `libc`. Since I'm already calling the Rust wrapper, I would probably just leave it like that.

https://docs.rs/libc/latest/aarch64-apple-darwin/libc/fn.makedev.html

Reply:
This may be worth filing a bug upstream with libc because this seems like it is a problem waiting to happen.

Reply:
@jbaublitz Ok, it's actually more interesting than that, because if you look at the `<sys/types.h>` code, makedev is just defined as a macro so the argument types can be anything really. In that sense, any type-safe non-generic wrapper changes the contract no matter what.

```
#define makedev(x, y)    ((dev_t)(((x) << 24) | (y)))
```

Reply:
Does that mean that there is a C stub in the libc crate that is calling the macro?

Reply:
It's actually inlined.

```
    pub {const} fn makedev(major: i32, minor: i32) -> dev_t {
        (major << 24) | minor
    }

    pub {const} fn major(dev: dev_t) -> i32 {
        (dev >> 24) & 0xff
    }

    pub {const} fn minor(dev: dev_t) -> i32 {
        dev & 0xffffff
    }
```

Reply:
How would you like to approach this? As long as I can tag you if something breaks with this support, I'm willing to go with what rust's libc reports as long as you can link an upstream issue here with libc so we can confirm that this is the desired behavior.

Reply:
Works for me, I'll create the issue.

Reply:
https://github.com/rust-lang/libc/issues/4360